cc_corpus/utils.py

```diff
@@ -349,5 +350,18 @@ def is_empty(s: Union[bytes, str]) -> bool:
     return s is None or not s.strip()
 
 
+def ensure_dir(directory: Union[Path, str]):
```

Comment:
I don't think we need this function at all.

First of all, just a minor point: if you are inside a "p is not a dir" clause, I see no point in writing exist_ok=True. It cannot be true at that point.

A more substantial problem is that in case of a Path input, this method is exactly the same as simply writing `p.mkdir(parents=True, exist_ok=True) isn't it?`

If you are worried about getting a str instead of a Path, because of the legacy usage of os.path, well, we were getting rid of those already, weren't we?
But if you are still worried, simply write:
`Path(p).mkdir(parents=True, exist_ok=True)`

Reply:
OK, I'll give you this one.